### PR TITLE
Handle BLEDevice name being `None` during discovery

### PR DESCRIPTION
**Pull request by bdraco**
will fix (but does not close) https://github.com/home-assistant/core/issues/96258

### CODE REVIEW

**Review by Jc2k — APPROVED** [review]

File: tests/test_controller_ble.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
```
Comment (bdraco) [inline tests/test_controller_ble.py:1]: Might as well start somewhere with the tests here

**Comment by bdraco** [comment]
Doing an unpair - repair to be sure everything is ok and than will merge + release

**Comment by bdraco** [comment]
All works but need to run to airport so will release when I get back from dropping off

**Comment by bdraco** [comment]
All good after retesting

**Pull request merged**